netty/src/test/java/io/grpc/netty/NettyChannelBuilderTest.java

```diff
@@ -90,6 +93,24 @@ public void failInvalidAuthority() {
   }
 
   @Test
+  public void sslContextCanBeNull() {
+    NettyChannelBuilder builder = mock(NettyChannelBuilder.class);
+    builder.sslContext(null);
```

Comment:
Carl, I'm unit testing the NettyChannelBuilder::sslContext() method. If I were changing the method like this :
public final NettyChannelBuilder sslContext(SslContext sslContext) {
    if ( sslContext.isServer()) { /* removing "sslContext != null &&"*/
      throw new IllegalArgumentException("Server SSL context can not be used for client channel");
    }
    this.sslContext = sslContext;
    return this;
  }

The unit test with mock object does throw null pointer exception:
io.grpc.netty.NettyChannelBuilderTest > sslContextCanBeNull FAILED
    java.lang.NullPointerException
        at io.grpc.netty.NettyChannelBuilder.sslContext(NettyChannelBuilder.java:173)
        at io.grpc.netty.NettyChannelBuilderTest.sslContextCanBeNull(NettyChannelBuilderTest.java:98)

mock(NettyChannelBuilder.class) for the concrete class seems to create an instance of an extension class of NettyChannelBuilder, and all non-abstract/final methods are not overridden with dummy return value.